.github/workflows/test.yaml

```diff
@@ -27,31 +43,46 @@ jobs:
         platform:
-          - [ubuntu22.04 , "linux/amd64,linux/arm64"]
-          - [ubuntu20.04 , "linux/amd64,linux/arm64"]
-          - [ubuntu18.04 , "linux/amd64,linux/arm64"]
-          - [ubuntu16.04 , "linux/amd64,linux/arm64"]
-          - [debian12    , "linux/amd64,linux/arm64"]
-          - [debian11    , "linux/amd64,linux/arm64"]
-          - [debian10    , "linux/amd64,linux/arm64"]
-          - [debian9     , "linux/amd64,linux/arm64"]
-          - [el9         , "linux/amd64,linux/arm64"]
-          - [el8         , "linux/amd64,linux/arm64"]
-          - [el7         , "linux/amd64,linux/arm64"]
-          - [amzn2       , "linux/amd64,linux/arm64"]
-          - [amzn2023    , "linux/amd64,linux/arm64"]
-          - [alpine3.15.1, "linux/amd64,linux/arm64"]
+          - [ubuntu22.04, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [ubuntu22.04, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [ubuntu20.04, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [ubuntu20.04, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [ubuntu18.04, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [ubuntu18.04, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [ubuntu16.04, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [ubuntu16.04, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [debian12, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [debian12, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [debian11, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [debian11, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [debian10, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [debian10, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [debian9, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [debian9, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [el9, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [el9, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [el8, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [el8, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [el7, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [el7, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [amzn2, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [amzn2, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [amzn2023, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [amzn2023, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
+          - [alpine3.15.1, linux/amd64, [self-hosted, linux, x64, ephemeral]]
+          - [alpine3.15.1, linux/arm64, [self-hosted, linux, arm64, ephemeral]]
 
     services:
       registry:
```

Comment:
I think we can remove this. It was needed because `docker/build-push-action` could not import multi-arch images automatically into docker context. So in order to use base images on the next step, we had to use local registry. Now since we do architecture-specific builds, the registry is not required anymore.